inngest-spring-boot-demo/src/main/java/com/inngest/springbootdemo/DevServerComponent.java

```diff
@@ -14,24 +15,25 @@ public class DevServerComponent {
     static String baseUrl = "http://127.0.0.1:8288";
     OkHttpClient httpClient = new OkHttpClient();
 
-    DevServerComponent() throws Exception {
+    DevServerComponent(CommHandler commHandler) throws Exception {
         Runtime rt = Runtime.getRuntime();
         rt.exec("pkill inngest-cli");
         rt.exec("npx -y inngest-cli dev -u http://127.0.0.1:8080/api/inngest");
 
-        waitForStartup();
+        waitForStartup(commHandler);
     }
 
-    private void waitForStartup() throws Exception {
-
+    private void waitForStartup(CommHandler commHandler) throws Exception {
         while (true) {
             try {
                 Request request = new Request.Builder()
-                    .url(baseUrl)
+                    .url(String.format("%s/health", baseUrl))
                     .build();
 
                 try (Response response = httpClient.newCall(request).execute()) {
                     if (response.code() == 200) {
+                        Thread.sleep(3000);
+                        commHandler.register("http://localhost:8080");
```

Comment:
possible future improvement: it looks like this sleeps 3 seconds regardless, but it'd be nice if `commHandler.register` would try multiple times up to 3 seconds so not every test is slowed down a fixed amount

Reply:
note: we also tried without the `sleep` and the errors were back so we'll keep it for now and dig into the issue more later.